File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func averageOfLevels(root *TreeNode) []float64 {
    if root == nil{
        return []float64{}
    }
    curLevelQ := []*TreeNode{root}//計算每階層數量
    averagePerLevel := []float64{}//返回的結果，每階層如何
    for len(curLevelQ) != 0{
        
        nextLevelQ := []*TreeNode{}
        curLevelSum := 0.0
        
        for _, node:= range curLevelQ{
        
            // update current level summation
            if node != nil{
                curLevelSum += float64(node.Val)
            }
            if node.Left != nil{
                nextLevelQ = append(nextLevelQ, node.Left)
            }
            if node.Right != nil{
                nextLevelQ = append(nextLevelQ, node.Right)
            }
        }
        curAvg := curLevelSum / float64(len(curLevelQ))
        averagePerLevel = append(averagePerLevel,curAvg)
        curLevelQ = nextLevelQ
    }
    return averagePerLevel
}

File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root):
        """
        :type root: TreeNode
        :rtype: List[float]
        """
        if root is None:
            return []
        
        result = []
        current_level = [root]
        while current_level:
            level_nodes = []
            next_level = []
            
            for node in current_level:
                level_nodes.append(node.val)
                if node.left:
                    next_level.append(node.left)
                if node.right:
                    next_level.append(node.right)
            
            result.append(sum(level_nodes)/len(level_nodes))
            current_level = next_level
        return result        
        